### ISSUE TITLE
Add support for signing models

### ISSUE THREAD

**Issue by rogancarr** [dotnet/machinelearning#1916]
When a model is passed to a third party, it would be nice to have a guarantee for where the model originated, and that its parameters have not been modified. I propose adding a method for signing a model in such a way that the original developer can be identified an the contents can be shown to be unchanged.

Related to #511

**Comment by rosebyte** [comment]
triage: we would need to see more demand for this feature, if anyone has a real-world use case blocked by the lack of signing, please let us know.
